code/game/gamemodes/umbrage/umbrage_objects.dm

```diff
@@ -13,10 +13,12 @@
 /obj/item/weapon/dark_bead/New()
 	..()
 	animate(src, alpha = 30, time = 30)
-	spawn(30)
-		if(!eating)
-			loc << "<span class='warning'>You were too slow! [src] faded away.</span>"
-			qdel(src)
+	QDEL_IN(src, 30)
+
+/obj/item/weapon/dark_bead/Destroy()
+	if(isliving(loc) && !eating)
```

Comment:
QDEL_IN just calls qdel, and the bead can't be destroyed if it's eating someone. It seems to work fine in testing.